### PR TITLE
Use common-utils instead of shajs in odsp-driver

### LINKED ISSUE #6999: odsp-driver package is bringing in a dependency of "buffer" package
odsp-driver package is bringing in a dependency of "buffer" package of size ~6kb gzipped. odsp-driver any of its dependencies does not seem to have any direct dependency on "buffer". Could you check if it is indeed required?

### PR DESCRIPTION
**Pull request by heliocliu**
Fixes #6999, next part of PR #7007

Use the updated hash functions from common-utils (with new SHA256/base64 support) instead of sha.js, which removes the sha.js and downstream dependencies and cuts ~37KB from the odsp-driver package.  Requires making `getHashedDocumentId` async.

### CODE REVIEW

**Comment by heliocliu** [comment]
This doesn't pass validation atm but will once #7007 is merged

**Comment by jatgarg** [comment]
Does it make the same hashedDocid as before?

**Comment by heliocliu** [comment]
> Does it make the same hashedDocid as before?

Yes, #7007 adds tests for the modified common-utils APIs and odsp-driver tests (which exercise the API) still pass with both changes

File: api-report/odsp-driver.api.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ export function encodeOdspFluidDataStoreLocator(locator: OdspFluidDataStoreLocat
 export function getApiRoot(origin: string): string;
 
 // @public (undocumented)
-export function getHashedDocumentId(driveId: string, itemId: string): string;
+export function getHashedDocumentId(driveId: string, itemId: string): Promise<string>;
```
Comment (vladsud) [inline api-report/odsp-driver.api.md:41]: Please take a look at usage in Bohemia. I believe they are using it and logging as well, such that their events have same structure and we can do queries across tables

**Review by vladsud — APPROVED** [review]

**Pull request merged**